### PR TITLE
Added support for passing delete options when using kubectlDelete.#3463

### PR DESCRIPTION
**Pull request by callingmahendra**
closes: #3463 

Added an option to pass deleteOptions in kubectlDelete.

### CODE REVIEW

File: extended/src/test/java/io/kubernetes/client/extended/kubectl/KubectlDeleteTest.java [hunk 1]
```diff
@@ -207,6 +209,9 @@ void kubectlDelete() throws KubectlException, ApiException {
         KubectlDelete<V1Job> kubectlDelete = Kubectl.delete(V1Job.class);
         kubectlDelete.apiClient(apiClient);
         kubectlDelete.namespace("foo").name("bar");
+        DeleteOptions deleteOptions = new DeleteOptions();
+        deleteOptions.setPropagationPolicy("Foreground");
+        kubectlDelete.deleteOptions(deleteOptions);
```
Comment (brendandburns) [inline extended/src/test/java/io/kubernetes/client/extended/kubectl/KubectlDeleteTest.java:214]: Can you add validation that this worked correctly to the URL validation below

**Comment by brendandburns** [comment]
Thanks for the PR. I think you need to run `mvn spotless:apply` to perform formatting for style, and expand the unit test slightly.

File: extended/src/test/java/io/kubernetes/client/extended/kubectl/KubectlDeleteTest.java (hunk 1 above)
Comment (callingmahendra) [reply extended/src/test/java/io/kubernetes/client/extended/kubectl/KubectlDeleteTest.java:214]: Added check below for URL validation and you were correct it was not passing deleteOptions, hence GenericKubernetesApi.java to pass it.

File: util/src/main/java/io/kubernetes/client/util/generic/GenericKubernetesApi.java [hunk 2]
```diff
@@ -1159,6 +1159,7 @@ private CallBuilder makeNamespacedDeleteCallBuilder(
             namespace,
             this.resourcePlural,
             name)
+                .body(deleteOptions)
                 // TODO: fill/convert the option
```
Comment (callingmahendra) [inline util/src/main/java/io/kubernetes/client/util/generic/GenericKubernetesApi.java:1163]: Should we remove this TODO as this is no more valid?

**Comment by callingmahendra** [comment]
> Thanks for the PR. I think you need to run `mvn spotless:apply` to perform formatting for style, and expand the unit test slightly.

Done both the changes, please review

File: util/src/main/java/io/kubernetes/client/util/generic/GenericKubernetesApi.java (hunk 2 above)
Comment (brendandburns) [reply util/src/main/java/io/kubernetes/client/util/generic/GenericKubernetesApi.java:1163]: Yes, please delete this now.

**Comment by brendandburns** [comment]
Thanks for adding the test, and fixing the code.

I think it still needs formatting, and that TODO removed, but then I think it's good to merge.

**Comment by callingmahendra** [comment]
> Thanks for adding the test, and fixing the code.
> 
> I think it still needs formatting, and that TODO removed, but then I think it's good to merge.

Removed the TODO, and also rerun mvn spotless:apply to make sure formatting apply.

**Pull request merged**